internal/transport/http2_client.go

```diff
@@ -1512,52 +1535,30 @@ func (t *http2Client) operateHeaders(frame *http2.MetaHeadersFrame) {
 						"protocol error: informational header with status code %d must not have END_STREAM set", statusCode))
 					t.closeStream(s, se.Err(), true, http2.ErrCodeProtocol, se, nil, endStream)
 				}
+				// In case of informational headers return
```

Comment:
Lots of options, here, e.g.:

```suggestion
				// In case of informational headers, return.
```